src/cmd/sign.js

```diff
@@ -13,12 +13,46 @@ import {createLogger} from '../util/logger';
 const log = createLogger(__filename);
 export const extensionIdFile = '.web-extension-id';
 
-export default function sign(
-    {verbose, sourceDir, artifactsDir, apiKey, apiSecret,
-     apiUrlPrefix, id, timeout}: Object,
-    {build=defaultBuilder, signAddon=defaultAddonSigner,
-     preValidatedManifest=null}: Object = {}): Promise<Object> {
+// Flow Types
+
+import type {ExtensionManifest} from  '../util/manifest';
+
+export type SignCmdParams = {
+  id?: string,
+  verbose?: boolean,
+  sourceDir: string,
+  artifactsDir: string,
+  apiKey: string,
+  apiSecret: string,
+  apiUrlPrefix: string,
+  timeout: number,
+};
+
+export type SignCmdExtraParams = {
+  build?: typeof defaultBuilder,
+  signAddon?: typeof defaultAddonSigner,
+  preValidatedManifest?: ExtensionManifest,
+};
 
+export type SignCmdResult = {
```

Comment:
Yay, this makes me happy. I don't think we even had this documented in a single place before. Later on, we may want to add documenting comments to things like this since it could be consumed by an external web-ext library user.